src/goto-symex/goto_symex_state.cpp

```diff
@@ -280,6 +280,46 @@ static void set_l2_indices(
   ssa_expr.set_level_2(level2.current_count(ssa_expr.get_identifier()));
 }
 
+void goto_symex_statet::rename_level0(exprt &expr, const namespacet &ns)
```

Comment:
I would expect that adding this function goes together with removing the same code from another function. As-is, it's difficult to assess whether this code does the same as the old one.